drivers/mfd/mfd_adi_max6639.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2025 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_max6639
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mfd/adi_max6639.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(max6639_mfd, CONFIG_MFD_LOG_LEVEL);
+
+struct max6639_config {
+	struct i2c_dt_spec i2c;
+};
+
+static int max6639_init(const struct device *dev)
+{
+	const struct max6639_config *config = dev->config;
+	int result;
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
```

Comment:
This is a little bit interesting. It's also not super obvious to me at the moment how customary this is, but usually the init system will never initialize a child node without already having initialized the parent node.

So again, not blocking, I just am not sure if this is necessary.